frontend/src/lib/hooks/use-auth.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState, createContext, useContext, type ReactNode } from 'react';
+import { useSession } from 'next-auth/react';
+import { type User, UserAPI } from '@api/user';
+import { isErrorMessage, type ErrorMessage } from '@utils/error';
+
+interface HookObject {
```

Comment:
Kanskje kalle denne for noe annet?

```suggestion
interface AuthInfo {
```